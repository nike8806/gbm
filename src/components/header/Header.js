import React, { Component } from 'react';
import { auth } from 'firebase';
import PropTypes from 'prop-types';
import Loader from '../loader/Loader';
import ROUTES from '../../constants/routes';
import './Header.scss';

/**
 * Handle the Header page
 */
class Header extends Component {
  static propTypes = {
    history: PropTypes.object.isRequired,
    children: PropTypes.node.isRequired
  };

  state = {
    showLoader: false
  }

  /**
  * OnSubmit function to try logout with Firebase
  */
  onSubmitLogout = () => {
    const { history } = this.props;

    this.setState({ showLoader: true });
    auth()
      .signOut()
      .then((response) => {
        this.setState({ showLoader: false });
        history.push(ROUTES.LOGIN);
        console.log(response);
      }).catch((error) => {
        // Handle Errors here.
        const errorCode = error.code;
        const errorMessage = error.message;
        console.log(errorCode, errorMessage);
        this.setState({ showLoader: false });
      });
  }

  render() {
    const { showLoader } = this.state;
    const { children } = this.props;

    return (
      <div className="header">
        <div className="header__title">
          <input
            type="button"
            onClick={this.onSubmitLogout}
            value="Logout"
          />
        </div>

        <div className="header__title">
          {children}
        </div>
        {showLoader && (
          <Loader />
        )}
      </div>
    );
  }
}

export default Header;
