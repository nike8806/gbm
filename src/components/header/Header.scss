
.header {
  &__container {
    display: block;
    margin: 0 auto;
  }
  h1 {
    margin: 20px;
    padding: 25px;
    line-height: 30px;
    font-size: 25px;
    font-weight: 300;
    color: #f7f7f7;
    text-align:center;
  }

  #button-logout{
    width: 20%;
    height: 50px;
    padding: 0;
    font-size: 20px;
    color: #fff;
    text-align: center;
    background: #f0776c;
    border: 0;
    border-radius: 5px;
    cursor: pointer; 
    outline:0;
    float: right;
    position: relative;
  }
}
