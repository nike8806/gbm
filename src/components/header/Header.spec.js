import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Header from './Header';

let component;

describe('Header component', () => {
  describe('On mount', () => {
    beforeEach(() => {
      const history = {};
      const children = <div>Something</div>;
      component = shallow(
        <Header history={history} children={children}/>
      );
    });

    it('should render correctly', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
