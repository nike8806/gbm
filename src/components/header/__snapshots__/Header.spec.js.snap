// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component On mount should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    history={Object {}}
  >
    <div>
      Something
    </div>
  </Header>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="header__title"
        >
          <input
            onClick={[Function]}
            type="button"
            value="Logout"
          />
        </div>,
        <div
          className="header__title"
        >
          <div>
            Something
          </div>
        </div>,
        false,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <input
            onClick={[Function]}
            type="button"
            value="Logout"
          />,
          "className": "header__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onClick": [Function],
            "type": "button",
            "value": "Logout",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div>
            Something
          </div>,
          "className": "header__title",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Something",
          },
          "ref": null,
          "rendered": "Something",
          "type": "div",
        },
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="header__title"
          >
            <input
              onClick={[Function]}
              type="button"
              value="Logout"
            />
          </div>,
          <div
            className="header__title"
          >
            <div>
              Something
            </div>
          </div>,
          false,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              onClick={[Function]}
              type="button"
              value="Logout"
            />,
            "className": "header__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onClick": [Function],
              "type": "button",
              "value": "Logout",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div>
              Something
            </div>,
            "className": "header__title",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Something",
            },
            "ref": null,
            "rendered": "Something",
            "type": "div",
          },
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
